src/Databases/PostgreSQL/DatabaseMaterializedPostgreSQL.cpp

```diff
@@ -128,6 +131,42 @@ void DatabaseMaterializedPostgreSQL::loadStoredObjects(
 }
 
 
+void DatabaseMaterializedPostgreSQL::checkAlterIsPossible(const AlterCommands & commands, ContextPtr) const
+{
+    for (const auto & command : commands)
+    {
+        if (command.type != AlterCommand::MODIFY_DATABASE_SETTING)
+            throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Alter of type '{}' is not supported by database engine {}", alterTypeToString(command.type), getEngineName());
+    }
+}
+
+
+void DatabaseMaterializedPostgreSQL::tryApplySettings(const SettingsChanges & settings_changes, ContextPtr local_context)
+{
+    for (const auto & change : settings_changes)
+    {
+        if (!settings->has(change.name))
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Database engine {} does not support setting `{}`", getEngineName(), change.name);
+
+        if ((change.name == "materialized_postgresql_tables_list"))
```

Comment:
It looks like check, we have to move it to the method above.

Reply:
Removed the method above.